src/sage/geometry/cone_critical_angles.py

```diff
@@ -0,0 +1,998 @@
+r"""
+Find maximal angles between polyhedral convex cones
+
+Finding the maximal (or equivalently, the minimal) angle between two
+polyhedral convex cones is a hard nonconvex optimization problem. The
+problem for a single cone was introduced in [IS2005]_, and was later
+extended in [SS2016]_ to two cones as a generalization of the
+principal angle between two vector subspaces.
+
+Seeger and Sossa proposed an algorithm to find maximal angles, and
+[Or2020]_ elaborates on that algorithm. It is this latest improvement
+that is implemented (more or less) by this module. The fact that
+perturbations of pointed cones may not change the answer too much
+[Or2024]_ is taken into consideration to avoid pathological cases.
+
+This module is internal to SageMath; the interface presented to users
+consists of a public method,
+:meth:`sage.geometry.cone.ConvexRationalPolyhedralCone.max_angle` for
+polyhedral convex cones. Even though all of the functions in this
+module are internal, some are more internal than others. There are a
+few functions that are used only in doctests, and not by any code that
+an end-user would run. Breaking somewhat with tradition, only those
+methods have been prefixed with an underscore.
```

Comment:
I don't understand the last sentence. Although there are benefits to exposing the functions to the public doc, even for technical functions. They still wouldn't be accessible from Sage's API. So I would probably not make them hidden in general (unless perhaps they are just a trivial helper function).

Reply:
Typically a module will have functions that are part of the public API, and internal (underscore) methods that are not. But in this case, the entire module should be considered internal:

> This module is internal to SageMath

Moreover it contains an extra class of functions beyond just the public and internal ones, namely functions used only for testing. Those testing-only functions are certainly internal, so I've given them an underscore prefix. But now what should we do with the non-testing functions, all of which are *also* internal?

If we give them all underscores, then there's no way to differentiate between the testing-only functions and the ones that are called in real code by way of `max_angle()`. So, in this case, I've left the underscore off the internal functions that are used in real code.

Nothing is truly internal in python anyway so this seemed like the least-bad option to me. I'm open to suggestions though.

Reply:
Okay, thanks. Now I understand.

I would probably lean towards making them all visible so they appear in the documentation. That way someone looking at this module's doc will know they are there and could potentially use them as part of their own testing. It seems reasonable to expose them in the documentation (especially considering how much effort you spent documenting these hidden functions). Anyways, it's okay to leave it the way it is too. I don't have a very strong opinion on this.

Reply:
They won't appear in the documentation anyway unless we add the module to the doc index (see the reverted TODO in the PR description). I personally like seeing the docs but I don't want to give anyone the impression that these are supported/stable interfaces. Not that I foresee major changes, but I'd like to reserve the right to hack away parts of  `cone_critital_angles.py` without deprecation warnings so long as `max_angle()` remains untouched.

Reply:
I would add it to the doc index. You're right that having it exposed like that would mean, strictly speaking, we should deprecate stuff. Although we can be flexible, but you can also add some redundancy with an additional statement saying everything except `max_angle()` in this file might change without a deprecation.

Reply:
I envy your optimism :)

I've added it back to the doc index along with a big warning at the top of the module doc stating that everything is subject to change.

I think that's it, thanks for taking a look at this.